% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\usepackage[utf8]{inputenc}
\DeclareUnicodeCharacter{00A0}{\nobreakspace}
\usepackage[T1]{fontenc}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}


\title{G4Analysis Documentation}
\date{February 23, 2012}
\release{0.1.0}
\author{Zhu H.}
\newcommand{\sphinxlogo}{}
\renewcommand{\releasename}{Release}
\makeindex

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\def\PYG@tok@gd{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\def\PYG@tok@gu{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\def\PYG@tok@gt{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.25,0.82}{##1}}}
\def\PYG@tok@gs{\let\PYG@bf=\textbf}
\def\PYG@tok@gr{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\def\PYG@tok@cm{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\def\PYG@tok@vg{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@m{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@mh{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@cs{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\colorbox[rgb]{1.00,0.94,0.94}{##1}}}
\def\PYG@tok@ge{\let\PYG@it=\textit}
\def\PYG@tok@vc{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@il{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@go{\def\PYG@tc##1{\textcolor[rgb]{0.19,0.19,0.19}{##1}}}
\def\PYG@tok@cp{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@gi{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\def\PYG@tok@gh{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\def\PYG@tok@ni{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\def\PYG@tok@nl{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\def\PYG@tok@nn{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\def\PYG@tok@no{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\def\PYG@tok@na{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@nb{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@nc{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\def\PYG@tok@nd{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\def\PYG@tok@ne{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@nf{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\def\PYG@tok@si{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\def\PYG@tok@s2{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@vi{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@nt{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\def\PYG@tok@nv{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@s1{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@gp{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\def\PYG@tok@sh{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@ow{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@sx{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\def\PYG@tok@bp{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@c1{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\def\PYG@tok@kc{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@c{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\def\PYG@tok@mf{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@err{\def\PYG@bc##1{\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{##1}}}
\def\PYG@tok@kd{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@ss{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\def\PYG@tok@sr{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\def\PYG@tok@mo{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@mi{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@kn{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@o{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\def\PYG@tok@kr{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@s{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@kp{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@w{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\def\PYG@tok@kt{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\def\PYG@tok@sc{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@sb{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@k{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@se{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@sd{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}

\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\begin{document}

\maketitle
\tableofcontents
\phantomsection\label{index::doc}


The MDPackage is a program package write in python mostly base on the
MDAnalysis Pacakge. The aim of this package is for editing the input and
output files of MD software like Gromacs, Amber, and NAMD. It also used for
reading and analyzing the MD trajectory, which mostly in binary format and
cannot read directly.


\chapter{Parallel\_analysis}
\label{documentation_pages/Parallel_analysis:welcome-to-mdpackage-s-documentation}\label{documentation_pages/Parallel_analysis::doc}\label{documentation_pages/Parallel_analysis:parallel-analysis}

\section{Introduction}
\label{documentation_pages/Parallel_analysis:introduction}
Parallel analysis.py is used for calculate the distance and angle between two
bases groups. usually a group contain 1, 2 or 4 bases in a plane.
The angle is useful to analysis the base stack. Two stack bases usually have a
small angle and
uctuation.
If the opition ``--rmsd'' used, only one bases group will be selected and the RMSD
in z-axis for this group will be calculated.


\section{Usage}
\label{documentation_pages/Parallel_analysis:usage}
\textbf{Files}

\begin{tabulary}{\linewidth}{|L|L|L|L|}
\hline
\textbf{
Option
} & \textbf{
Type
} & \textbf{
Filename
} & \textbf{
Description
}\\
\hline

-p
 & 
Input
 & 
coor\_file
 & 
Structure fie: gro pdb etc.
\\

-f
 & 
Input
 & 
traj\_file
 & 
Trajectory: xtc trr.
\\

-o
 & 
Input
 & 
output\_file
 & 
xvgr/xmgr file.
\\

-i
 & 
Input
 & 
para\_an.in
 & 
input parmarter file.
\\
\hline
\end{tabulary}


\textbf{Other options}

\begin{tabulary}{\linewidth}{|L|L|L|L|}
\hline
\textbf{
Option
} & \textbf{
Type
} & \textbf{
Value
} & \textbf{
Description
}\\
\hline

--rmsd
 & 
bool
 & 
False
 & 
skip Calculate the RMSD of DNA bases groups.
\\

--skip
 & 
int
 & 
1
 & 
Get frames when frame MOD skip = 0
\\

-h
 & 
bool
 & 
yes
 & 
Print help info and quit
\\
\hline
\end{tabulary}



\section{Some details of the Algorithm}
\label{documentation_pages/Parallel_analysis:some-details-of-the-algorithm}

\chapter{Modify\_Coor}
\label{documentation_pages/Modify_Coor::doc}\label{documentation_pages/Modify_Coor:modify-coor}

\section{Introduction}
\label{documentation_pages/Modify_Coor:introduction}
Modify\_Coor convert a structure file from one format to another or modified
the structure file. The structure file can be in pdb, gro or pqr format.


\section{Usage}
\label{documentation_pages/Modify_Coor:usage}

\subsection{Interactive Mode}
\label{documentation_pages/Modify_Coor:interactive-mode}
\textbf{Modify\_Coor.py} to run interactive mode.
Type help can see the list of the key words. They are listed below.

\begin{tabulary}{\linewidth}{|L|L|L|}
\hline

load
 & 
\textless{} filename \textgreater{}
 & 
: Load a structure file: gro pdb etc.
\\

list
 &  & 
: List the groups.
\\

\textless{}Enter\textgreater{}
 &  & 
: List the groups.
\\

delete
 & 
\textless{} groupname \textgreater{}
 & 
: Delete a group.
\\

splitres
 & 
\textless{} groupname \textgreater{}
 & 
: Split a group to residues.
\\

splitatom
 & 
\textless{} groupname \textgreater{}
 & 
: Split a group to atoms.
\\

save
 & 
\textless{} filename \textgreater{}
 & 
: Save the result to a structure file: gro pdb etc.
\\

history
 &  & 
: Print the input history.
\\

help
 &  & 
: Print help information.
\\

quit
 &  & 
: Quit
\\
\hline
\end{tabulary}


\textbf{load}

Now many structure file format are allowed to load, include gro, pdb, pqr and amber top+crd.

\textbf{save}

Only two types of the structure file format (pdb and gro) can be saved by this program.


\subsection{Commond Mode}
\label{documentation_pages/Modify_Coor:commond-mode}
\textbf{Modify\_Coor.py -f filename -o filename} to run interactive mode.


\section{Some Detials}
\label{documentation_pages/Modify_Coor:some-detials}

\chapter{Trj\_modify}
\label{documentation_pages/Trj_modify::doc}\label{documentation_pages/Trj_modify:trj-modify}

\section{Introduction}
\label{documentation_pages/Trj_modify:introduction}
Trj\_modify is design for fit the trajectory to the first frame by translation and rotation. First it move the solvate to the ceter, then it fit to the first frame of the trajectory.


\section{Usage}
\label{documentation_pages/Trj_modify:usage}
\textbf{Trj\_modify topology\_file  trajectory\_file  output\_trajectory\_file}

Input trajectory file can be in any format support by \textbf{MDAnalysis}, and the suggest output format of trajectory file is \emph{xtc},

\begin{notice}{warning}{Warning:}
This program will read the pbc condition and use the dimensions read from trajectory files. You should make sure the dimensions are right or it will create a wrong output trajectory file.
\end{notice}


\section{Some detials of the Algorithm}
\label{documentation_pages/Trj_modify:some-detials-of-the-algorithm}

\chapter{Least squares fitting procedures}
\label{documentation_pages/least_squares_fitting::doc}\label{documentation_pages/least_squares_fitting:least-squares-fitting-procedures}

\section{Algorithm}
\label{documentation_pages/least_squares_fitting:algorithm}
Define \textbf{S} matrix and \textbf{E} matrix which are N*3 matrix, and the N is the number of coordinates.
\textbf{s\_ave} and \textbf{e\_ave} are two vector which are the average of the cloumn of \textbf{S} matrix and \textbf{E} matrix. This process fitting the \textbf{S} coordinates to \textbf{E} coorinates.

First, construct a 3*3 covariance matrix \textbf{C} between \textbf{S} matrix and \textbf{E} matrix using the following formula:
\begin{gather}
\begin{split}C=\frac{1}{N-1}(S^T  E - \frac{1}{N} S^T  I  I^T  E)\end{split}\notag\\\begin{split}\end{split}\notag
\end{gather}
Here \textbf{I} is an N*1 column vector consisting of only ones.

From the nine elements of \textbf{C}, we subsequently generate the 4*4 real symmetric matrix \textbf{M} using the expression:
\begin{gather}
\begin{split}M=
\begin{vmatrix}
c_{11}+c_{22}+c_{33} & c_{23}-c_{32}         &  c_{31}-c_{13}        & c_{12}-c_{21} \\
c_{23}-c_{32}        & c_{11}-c_{22}-c_{33}  &  c_{12}+c_{21}        & c_{31}+c_{13} \\
c_{31}-c_{13}        & c_{12}+c_{21}         & -c_{11}+c_{22}-c_{33} & c_{23}+c_{32} \\
c_{12}-c_{21}        & c_{31}+c_{13}         &  c_{23}+c_{32}        & -c_{11}-c_{22}+c_{33} \\
\end{vmatrix}\end{split}\notag\\\begin{split}\end{split}\notag
\end{gather}
The ($q_1,q_2,q_3,q_4$) is the eigenvector corresponding to the largest eigenvalue of matrix \textbf{M}. Using the element of the largest eigenvector, the orientation matrix \textbf{R} can be established by equation below:
\begin{gather}
\begin{split}R=
\begin{vmatrix}
q_0q_0+q_1q_1-q_2q_2-q_3q_3  &  2(q_1q_2 - q_0q_3)           & 2(q_1q_3+q_0q_2) \\
2(q_2q_1+q_0q_3)             &  q_0q_0-q_1q_1+q_2q_2-q_3q_3  & 2(q_2q_3-q_0q_1) \\
2(q_3q_1-q_0q_2)             &  2(q_3q_2 + q_0q_1)           & q_0q_0-q_1q_1- q_2q_2+q_3q_3 \\
\end{vmatrix}\end{split}\notag\\\begin{split}\end{split}\notag
\end{gather}
The first column of matrix \textbf{R} corresponds to x-axis of the base, the second to y-axis, and the third to z-axis.

The origin coordinate of the base can be calculated using equation below:
\begin{gather}
\begin{split}O=E_{ave}-S_{ave}R^{T}\end{split}\notag\\\begin{split}\end{split}\notag
\end{gather}
The $E_{ave}$ is the vector of the average of experimental coordinates,
and the $S_{ave}$ is the vector corresponding to the average of the standard coordinates of base.

Now the fitting coordinates can be calculate from the rotation matrix \textbf{R} and origin \textbf{o} using the formula below:
\begin{gather}
\begin{split}F=SR^T+o\end{split}\notag\\\begin{split}\end{split}\notag
\end{gather}
The upper algorithm consulted the 3DNA software package.


\chapter{File format list}
\label{documentation_pages/file_format::doc}\label{documentation_pages/file_format:file-format-list}

\section{PDB}
\label{documentation_pages/file_format:pdb}

\subsection{Introduction to Protein Data Bank Format}
\label{documentation_pages/file_format:introduction-to-protein-data-bank-format}
Protein Data Bank (PDB) format is a standard for files containing atomic coordinates. Structures
deposited in the Protein Data Bank at the Research Collaboratory for Structural Bioinformatics (RCSB) are
written in this standardized format. The short description provided here will suffice for most users. How-
ever, those actually creating PDB files should consult the definitive description (see
\href{http://www.rcsb.org/pdb/info.html\#File\_Formats\_and\_Standards}{http://www.rcsb.org/pdb/info.html\#File\_Formats\_and\_Standards}).

The complete PDB file specification provides for a wealth of information, including authors, literature
references, and the identification of substructures such as disulfide bonds, helices, sheets, and active
sites. Users should bear in mind that modeling programs can be unforgiving of incorrect input formats.


\subsection{Description}
\label{documentation_pages/file_format:description}
\begin{tabulary}{\linewidth}{|L|L|L|L|L|}
\hline
\textbf{
Record Type
} & \textbf{
Columns
} & \textbf{
Data
} & \textbf{
Justfication
} & \textbf{
DataType
}\\
\hline

ATOM
 & 
1-4
 & 
``ATOM''
 & 
left
 & 
character
\\
 & 
7-11
 & 
Atom serial number
 & 
right
 & 
integer
\\
 & 
13-16
 & 
Atom name
 & 
left*
 & 
character
\\
 & 
17
 & 
Alternate location indicator
 &  & 
character
\\
 & 
18-20
 & 
Residue name
 & 
right
 & 
character
\\
 & 
22
 & 
Chain identifier
 &  & 
character
\\
 & 
23-26
 & 
Residue sequence number
 & 
right
 & 
integer
\\
 & 
27
 & 
Code for insertions of residues
 &  & 
character
\\
 & 
31-38
 & 
X orthogonal Angstrom coordinate
 & 
right
 & 
floating
\\
 & 
39-46
 & 
Y orthogonal Angstrom coordinate
 & 
right
 & 
floating
\\
 & 
47-54
 & 
Z orthogonal Angstrom coordinate
 & 
right
 & 
floating
\\
 & 
55-60
 & 
Occupancy
 & 
right
 & 
floating
\\
 & 
61-66
 & 
Temperature factor
 & 
right
 & 
floating
\\
 & 
73-76
 & 
Segment identifier (optional)
 & 
left
 & 
character
\\
 & 
77-78
 & 
Element symbol
 & 
right
 & 
character
\\
 & 
79-80
 & 
Charge (optional)
 &  & 
character
\\
\hline
\end{tabulary}



\section{PQR}
\label{documentation_pages/file_format:pqr}
This format is a modification of the PDB format which allows users to add charge
and radius parameters to existing PDB data while keeping it in a format amenable
to visualization with standard molecular graphics programs. The origins of the PQR
format are somewhat uncertain, but has been used by several computational biology
software programs, including MEAD and AutoDock. UHBD uses a very similar format
called QCD.

APBS reads very loosely-formatted PQR files: all fields are whitespace-delimited
rather than the strict column formatting mandated by the PDB format. This more
liberal formatting allows coordinates which are larger/smaller than \(\pm\) 999 Å.

APBS reads data on a per-line basis from PQR files using the following format:


\bigskip\hrule{}\bigskip

\begin{quote}

Field\_name Atom\_number Atom\_name Residue\_name Chain\_ID Residue\_number X Y Z Charge Radius
\end{quote}


\bigskip\hrule{}\bigskip


where the whitespace is the most important feature of this format. The fields are:
\begin{itemize}
\item {} 
Field\_name

A string which specifies the type of PQR entry and should either be ATOM or HETATM in order to be parsed by APBS.

\item {} 
Atom\_number

An integer which provides the atom index.

\item {} 
Atom\_name

A string which provides the atom name.

\item {} 
Residue\_name

A string which provides the residue name.

\item {} 
Chain\_ID

An optional string which provides the chain ID of the atom. Note chain ID support is a new feature of APBS 0.5.0 and later versions.

\item {} 
Residue\_number

An integer which provides the residue index.

\item {} 
X Y Z

3 floats which provide the atomic coordiantes.

\item {} 
Charge

A float which provides the atomic charge (in electrons).

\item {} 
Radius

A float which provides the atomic radius (in Å).

\end{itemize}

Clearly, this format can deviate wildly from PDB due to the use of whitespaces
rather than specific column widths and alignments. This deviation can be
particularly significant when large coordinate values are used. However, in order
to maintain compatibility with most molecular graphics programs, the PDB2PQR
program and the utilities provided with APBS (see the Parameterization section)
attempt to preserve the PDB format as much as possible.


\section{GRO}
\label{documentation_pages/file_format:gro}
Files with the gro file extension contain a molecular structure in Gromos87 format. gro files
can be used as trajectory by simply concatenating files. An attempt will be made to read a
time value from the title string in each frame, which should be preceded by `t=', as in the
sample below.

A sample piece is included below:


\bigskip\hrule{}\bigskip


MD of 2 waters, t= 0.0

6

1WATER  OW1    1   0.126   1.624   1.679  0.1227 -0.0580  0.0434

1WATER  HW2    2   0.190   1.661   1.747  0.8085  0.3191 -0.7791

1WATER  HW3    3   0.177   1.568   1.613 -0.9045 -2.6469  1.3180

2WATER  OW1    4   1.275   0.053   0.622  0.2519  0.3140 -0.1734

2WATER  HW2    5   1.337   0.002   0.680 -1.0641 -1.1349  0.0257

2WATER  HW3    6   1.326   0.120   0.568  1.9427 -0.8216 -0.0244

1.82060   1.82060   1.82060


\bigskip\hrule{}\bigskip


Lines contain the following information (top to bottom):
\begin{itemize}
\item {} 
title string (free format string, optional time in ps after `t=')

\item {} 
number of atoms (free format integer)

\item {} 
one line for each atom (fixed format, see below)

\item {} 
box vectors (free format, space separated reals), values: v1(x) v2(y) v3(z)
v1(y) v1(z) v2(x) v2(z) v3(x) v3(y), the last 6 values may be omitted (they will
be set to zero). Gromacs only supports boxes with v1(y)=v1(z)=v2(z)=0.

\end{itemize}

This format is fixed, ie. all columns are in a fixed position. Optionally (for now only
yet with trjconv) you can write gro files with any number of decimal places, the format
will then be n+5 positions with n decimal places (n+1 for velocities) in stead of 8 with
3 (with 4 for velocities). Upon reading, the precision will be inferred from the distance
between the decimal points (which will be n+5). Columns contain the following information
(from left to right):
\begin{itemize}
\item {} 
residue number (5 positions, integer)

\item {} 
residue name (5 characters)

\item {} 
atom name (5 characters)

\item {} 
atom number (5 positions, integer)

\item {} 
position (in nm, x y z in 3 columns, each 8 positions with 3 decimal places)

\item {} 
velocity (in nm/ps (or km/s), x y z in 3 columns, each 8 positions with 4 decimal places)

\end{itemize}

Note that separate molecules or ions (e.g. water or Cl-) are regarded as residues. If you want
to write such a file in your own program without using the GROMACS libraries you can use the
following formats:

C format
\begin{quote}

``\%5d\%5s\%5s\%5d\%8.3f\%8.3f\%8.3f\%8.4f\%8.4f\%8.4f''
\end{quote}


\section{XYZ}
\label{documentation_pages/file_format:xyz}
The XYZ file format is a chemical file format. There is no formal standard and several
variations exist, but a typical XYZ format specifies the molecule geometry by giving the
number of atoms with Cartesian coordinates that will be read on the first line, a comment
on the second, and the lines of atomic coordinates in the following lines. The file format
is used in computational chemistry programs for importing and exporting geometries. The
units are generally in Ångströms. Some variations include using atomic numbers instead of
atomic symbols, or skipping the comment line. Files using the XYZ format conventionally
have the .xyz extension.


\subsection{Format}
\label{documentation_pages/file_format:format}
The formatting of the .xyz file format is as follows:


\bigskip\hrule{}\bigskip


\textless{}number of atoms\textgreater{}

comment line

atom\_symbol1 x-coord1 y-coord1 z-coord1

atom\_symbol2 x-coord2 y-coord1 z-coord2

...

atom\_symboln x-coordn y-coordn z-coordn


\bigskip\hrule{}\bigskip



\subsection{Example}
\label{documentation_pages/file_format:example}
The methane molecule can be described in the XYZ format by the following:


\bigskip\hrule{}\bigskip


5

methane molecule (in {[}{[}Ångström{]}{]}s)

C        0.000000        0.000000        0.000000

H        0.000000        0.000000        1.089000

H        1.026719        0.000000       -0.363000

H       -0.513360       -0.889165       -0.363000

H       -0.513360        0.889165       -0.363000


\bigskip\hrule{}\bigskip



\section{amber TOP}
\label{documentation_pages/file_format:amber-top}

\section{amber CRD}
\label{documentation_pages/file_format:amber-crd}

\chapter{The modules of MDPackage}
\label{documentation_pages/modules::doc}\label{documentation_pages/modules:the-modules-of-mdpackage}

\section{atomlib}
\label{documentation_pages/modules/atomlib::doc}\label{documentation_pages/modules/atomlib:atomlib}

\subsection{Variables}
\label{documentation_pages/modules/atomlib:variables}
BASE\_AG\_LIST = {[}'N9', `C8', `N7', `C5', `C6', `N1', `C2', `N3'...

It's a list of atoms which used to fit the standard structure for the A and G base.

BASE\_CTU\_LIST = {[}'N1', `C2', `N3', `C4', `C5', `C6'{]}

It's a list of atoms which used to fit the standard structure for the C,T and U base.

RESIDUE\_NAME\_LIST = {[}'A', `T', `C', `G', `DA', `DT', `DC', `DG...

just for nucleic now.

BASE\_A\_DICT = \{`C2': (-1.912, 1.023, 0.0), `C4': (-1.267, 3.12...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Adenine base.

BASE\_C\_DICT = \{`C2': (-1.472, 3.158, 0.0), `C4': (0.837, 2.868...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Cytosine base.

BASE\_G\_DICT = \{`C2': (-1.999, 1.087, 0.0), `C4': (-1.265, 3.17...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Guanine base.

BASE\_T\_DICT = \{`C2': (-1.462, 3.135, 0.0), `C4': (0.994, 2.897...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Thymine base.

BASE\_U\_DICT = \{`C2': (-1.462, 3.131, 0.0), `C4': (0.989, 2.884...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Uracil base.

BASE\_A\_array = array({[}{[}-1.291, 4.498, 0. ...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Adenine base.

BASE\_C\_array = array({[}{[}-1.285, 4.542, 0. ...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Cytosine base.

BASE\_G\_array = array({[}{[} -1.28900000e+00, 4.55100000e+00, 0...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Guanine base.

BASE\_T\_array = array({[}{[}-1.284, 4.5 , 0. ...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Thymine base.

BASE\_U\_array = array({[}{[}-1.284, 4.5 , 0. ...

The cartesian coordinates of non-hydrogen atoms in the standard reference frames of the Uracil base.


\chapter{Change log}
\label{documentation_pages/change::doc}\label{documentation_pages/change:change-log}

\section{version 0.1.0}
\label{documentation_pages/change:version-0-1-0}

\subsection{Modules and Scripts}
\label{documentation_pages/change:modules-and-scripts}\begin{itemize}
\item {} 
Parallel\_analysis

\item {} 
Modified\_Coor

\end{itemize}


\subsubsection{Modules List}
\label{documentation_pages/change:modules-list}\begin{itemize}
\item {} 
atomlib

\end{itemize}


\chapter{Indices and tables}
\label{index:indices-and-tables}\begin{itemize}
\item {} 
\emph{genindex}

\item {} 
\emph{modindex}

\item {} 
\emph{search}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}
